Sources/GoodSwiftUI/ReadableContentWidthModifier.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  ReadableContentWidthModifier.swift
+//
+//  GoodSwiftUI
+//  Created by Filip Šašala on 31/12/2023.
+//
+
+import SwiftUI
+
+private struct ReadableContentWidth: ViewModifier {
+
+    private let measureViewController = UIViewController()
+
+    @State private var orientation: UIDeviceOrientation = UIDevice.current.orientation
+
+    func body(content: Content) -> some View {
+        content
+            .frame(maxWidth: readableWidth(for: orientation))
+            .onReceive(NotificationCenter.default.publisher(for: UIDevice.orientationDidChangeNotification)) { _ in
+                orientation = UIDevice.current.orientation
```

Comment:
You are going to receive this notification for all orientation cases it means even faceUp, faceDown etc. The best would be to listen for `viewWillTransitionToSize` inside each controller wrapper but don't see an ideal solution right now.

Maybe we can use some kind of introspection to obtain wrapped view controller and listen for this method change. On the onther hand, in case we can introspect controller we can automatically extract readableContentGuide size.